module-2/backend/week_one.md

```diff
@@ -5,30 +5,123 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+
 11. What's responsive design?
+
+ Responsive design is designing with a mobile-first strategy that yields a result that is responsive to the changing browser size. As the browser size changes, the content on the page shifts to be readable.
+
 12. What is CRUD and why is it important?
+
+  CRUD stands for Create, Read, Update, Delete. It's important because it represents everything we can do to data and the base functionality our app should have.
+  
 13. What does HTTP stand for? 
+
+ HyperText Transfer Protocol
+
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+
+ Renders on page: ``` <%= %>```
+ 
+ Is evaluated but not rendered: ``` <% %>```
+ 
 15. What's an ORM?
+
+ ORM stands for Object Relational Mapper. (I'm a bit fuzzy on the concept, but will do my best to explain without the documentation). I think an ORM is something that converts / is a middle man for Ruby objects and rows/tables in the database. This means that an individual row in a table is representative of the attributes of a Ruby object but is not a Ruby object itself. The ORM allows me to retrieve that row (instance) and puts a wrapper on it so I can interact with it with some of the Ruby API.
+
 16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+
+ ActiveRecord
+
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+
+ Create
+  * User can add a restaurant
+    * GET '/restaurants/new' (retrieves form)
+    * POST '/restaurants' (adds new restaurant to db, displays all data on index view)
+   
+ Read
+   * User can see all restaurants
+     * GET '/restaurants'
+   * User can see a single restaurant
+     * GET '/restaurants/:id' 
+ 
+ Update
+   * User can update an existing restaurant
+     * GET '/restaurants/:id/edit' (retrieves form)
+     * PUT '/restaurants/:id' (updates to db, displays new data on show view)
+ 
+ Delete
+   * User can delete a restaurant
+     * DELETE '/restaurants/:id' (deletes from db, displays index view)
+
 18. What's a migration? 
+
+  a process that adds structure to an existing database
+ 
 19. When you create a migration, does it automatically modify your database?
+
+ not unless ``` rake db:migrate ``` has been executed
+
 20. How does a model relate to a database?
+
+ A model is a representation of aspects of the database and tables within the database (or perhaps the other way around?). A model is a ruby object that, because it can inherit from ActiveRecrd can be represented by a database. Any attributes we would want this model to have as a Ruby object translate to columns of the table, and instantiations of the model's class are represented by rows of the table. The model also has behavior in that it can interact with other models via foreign keys in the database.

```

Comment:
I like the first way you describe this better than the second. To me, the database is the source of truth for our application, and the models are ruby objects that we use to interact with the database.

One other note: the interaction with other models isn't automatic once we have foreign keys. Foreign keys establish relationships at the database level. In order to take advantage of those relationships at the model level, we need to add our `has_many` and `belongs_to` methods to our models.